### ISSUE TITLE
OOM in Regex::new()

### ISSUE THREAD

**Issue by lukaslueg** [rust-lang-nursery/regex#254]
AFL found this simple regex which exceeds the default size limit of `Regex::new()`. Raising the upper repetition limit will outmatch any given size limit, consuming gigabytes of memory.

```
${1,999999}
```

**Comment by BurntSushi** [comment]
I can't seem to reproduce this. Could you provide a complete program?

If I run this program, then I get back the expected error: `CompiledTooBig`.

```rust
extern crate regex;

use regex::Regex;

fn main() {
    let re = Regex::new(r"${1,999999999}").unwrap();
    println!("{:?}", re.find("foo"));
}
```

**Comment by BurntSushi** [comment]
@lukaslueg Nope, `{m,n}` aren't loops, they are plain repeats. Having `a{10}` is exactly equivalent to `aaaaaaaaaa`.
